docs/source/security/cors.mdx

```diff
@@ -0,0 +1,108 @@
+
+## Configuring CORS options for Apollo Server
+
+The batteries-included `apollo-server` [has a `cors` option](../api/apollo-server/#cors), which you can use to change your server's CORS behavior:
+
+```js
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  cors: {
+    origin: ["http://localhost:3000", "https://studio.apollographql.com"]
+  },
+});
+```
+
+If you're using another integration of Apollo Server, you can add [CORS configuration options](../api/apollo-server/#cors-1) to your framework-specific middleware function.
+
+Below is an example of setting up CORS with the [`apollo-server-express`](../integrations/middleware#swapping-out-apollo-server) package:
+
+```js
+// ... set up the server
+await server.start();
+
+const corsOptions = {
+  origin: ["http://localhost:3000", "https://studio.apollographql.com"]
+};
+
+server.applyMiddleware({
+  app,
+  cors: corsOptions,
+  path: "/graphql",
+});
+```
+
+### Configuring origins
+
+CORS uses [specific HTTP request headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers) as part of its protocol; the most well-known is [`Access-Control-Allow-Origin`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin). The `Access-Control-Allow-Origin` header enables a server to dictate which origins can use scripts to access that server's resources. Using [`cors`](https://github.com/expressjs/cors#configuration-options) middleware, you can configure the `Access-Control-Allow-Origin` header using the `origins` option.
+
+By default, the batteries-included `apollo-server` package uses the wildcard value (`*`) to allow requests from any origin _without credentials_ to access its served responses. This works great in development, allowing you to delay configuring CORS until your application is ready for production. You can also [remove CORS middleware entirely](../api/apollo-server#cors), though allowing all traffic to your server can lead to security implications.
+
+> If you need to pass credentials to your server (e.g., via cookies or an HTTP header), you can't use the wildcard value (`*`) for your origin. You _must_ provide specific origins; see [Passing credentials with CORS](#passing-credentials-with-cors) for examples.
+
+When your server is in production, we strongly recommend listing the specific origins that should have access to your server:
+
+```js
+//  With the batteries-included Apollo Server
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  cors: {
+    origin: ["http://localhost:3000", "https://studio.apollographql.com"]
+  },
+});
+```
+
+> Note if you plan to use Apollo Studio you should include https://studio.apollographql.com in your list of valid origins.
```

Comment:
Specifically, this is if you want to use Apollo Studio Explorer as a GraphQL web IDE (probably link to https://www.apollographql.com/docs/studio/explorer/explorer/ or something).

Reply:
Anywhere a URL isn't a link such as here, let's put it in `codefont`